components/icon-button/index.js

```diff
@@ -16,21 +16,56 @@ import Tooltip from '../tooltip';
 import Button from '../button';
 import Dashicon from '../dashicon';
 
-// This is intentionally a Component class, not a function component because it
-// is common to apply a ref to the button element (only supported in class)
 class IconButton extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.dismissTooltip = this.dismissTooltip.bind( this );
+		this.resetClicked = this.resetClicked.bind( this );
+
+		this.state = {
+			isClicked: false,
+		};
+	}
+
+	dismissTooltip() {
```

Comment:
When I first looked at the changed, I was wondering where this prop was being handled because when you read

`onClick={ this.dismissTooltip }` you do not assume this would call the onClick prop. I wonder if it's a good idea to rename this function maybe just `onClick`